learningobjects/api.py

```diff
@@ -0,0 +1,39 @@
+"""
+Helper functions for using the models, so external
+apps don't tie functionality to internal implementation.
+"""
+
+from datetime import datetime
+
+from learningobjects.models import Course, Repository
+
+# pylint: disable=no-member
+
+def create_course(org, course_number, semester, user_id):
+    """
+    Add a course to the database.
+    """
+    # TODO: Determine rules for accepting/requiring a repository.
+    kwargs = {
+        "org": org, "course_number": course_number, "semester": semester,
+        "import_date": datetime.now(), 'imported_by_id': user_id,
+        "repository_id": get_temp_repository(user_id).id
+    }
+    courses = Course.objects.filter(**kwargs)
+    if courses.count() > 0:
+        return None, "course already exists"
+    return Course.objects.create(**kwargs), ""
```

Comment:
Ok, so we need to decide on this one way or the other... I thought @ShawnMilo were specifically recommending  not using models, but several of the use cases you suggested for the API approach could happen if we are returning models and caller functions could/will become dependent on model coming back and the fields and methods therein. Sounds like we need a powow